browser/src/GenePage/GenePage.spec.tsx

```diff
@@ -1,30 +1,142 @@
 import { jest, describe, expect } from '@jest/globals'
 import React from 'react'
 import renderer from 'react-test-renderer'
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
 
 import geneFactory from '../__factories__/Gene'
 import GenePage from './GenePage'
 import { DatasetId } from '@gnomad/dataset-metadata/metadata'
-import { apiCallsMatching } from '../../../tests/__helpers__/apiCall'
-import { forAllDatasets } from '../../../tests/__helpers__/datasets'
+import { forDatasetsMatching, forDatasetsNotMatching } from '../../../tests/__helpers__/datasets'
+import { mockQueries } from '../../../tests/__helpers__/queries'
 import { withDummyRouter } from '../../../tests/__helpers__/router'
 
-forAllDatasets('GenePage with dataset "%s"', (datasetId) => {
-  const fetch = jest.fn(() => {
-    return new Promise(() => {})
+import Query, { BaseQuery } from '../Query'
```

Comment:
Sure--if it would help clarify this PR, I could rebase things around so that at least this helper (and the other new one) are in separate commits. Then you could look at each of those commits in isolation, at least.